### DISCUSSION TITLE
Setting commitWhenAutocommitDisabled for SchedulerClient

### DISCUSSION THREAD

**Discussion by ot-yd** [kagkarlsson/db-scheduler#506] [Q&A] [answered]
Similar discussions: #351, #146, #287, one of which introduces `commitWhenAutocommitDisabled`  for `SchedulerBuilder`.
Question: Is there a reason to not have it in `SchedulerClient` / `SchedulerClient.Builder`?

My use-case:
I'm using `SchedulerClient` to submit tasks. The `DataSource` used by `SchedulerClient` is a pooled `DataSource`, so a `Connection` given to `JdbcTaskRepository` by such a data source may have `autoCommit` set to either true or false (set by previous `Connection` user). When it happens to get `Connection` with `autoCommit == false`, the task will not be created.

I can proxy a data source to set `autoCommit` to true, but this might create problems in future if transactional `CompletionHandler`'s are implemented.

Thanks

**Comment by kagkarlsson** [comment]
We can propbably add it. I always forget why we need it. If someone sets `autoCommit == false`, then it should be their resposibility to do the final commit. If the connection-pool is handing out connections with autocommit=false and not doing any commit on return to pool, then I suppose we need this setting

**Reply by ot-yd** [reply]
It's a pool that ~~is not externally managed~~ does not manage connections externally, it just hands out connections. I believe there is no way for a `Task` or `CompletionHandler` to obtain a connection to commit it. I also don't think there is an easy way to commit it without existing external transaction management.

**Reply by kagkarlsson** [reply, marked as answer]
Ok, thanks. I makes sense to add it to ScheduleClient as well, since we have it in the SchedulerBuilder

**Reply by ot-yd** [reply]
Thank you
